### PR TITLE
[ACEAO-4923] add prerender for seo

### PR DESCRIPTION
**Pull request by vokidya**
`curl https://teksystems-staging-api.cbtalentnetwork.com/en-US/search`

`curl -A Googlebot https://teksystems-staging-api.cbtalentnetwork.com/en-US/search`

And view the response HTML, you will get the Angular {{}} rendered for the second one.

### CODE REVIEW

File: include/http/middleware/prerender.js [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+        } catch(ex) {
+            // Just go normally path if there is any errors happen.
+            console.log(ex);
+        }
+    }
+});
+
+const extensionsToIgnore = [
+    '.js',
```
Comment (CB-JackLi) [inline include/http/middleware/prerender.js:35]: emm but is this list referenced from somewhere? (I guess so)

Comment (vokidya) [reply include/http/middleware/prerender.js:35]: https://github.com/RisingStack/koa-prerender

File: include/http/middleware/prerender.js [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+}
+
+function getPrerenderedPage(req) {
+    var options = {
+        uri: url.parse(buildApiUrl(req)),
+        followRedirect: false,
+        headers: {}
+    };
+
+    Object.keys(req.headers).forEach(function (h) {
```
Comment (cecilliaQin) [inline include/http/middleware/prerender.js:153]: clean code

**Review by cecilliaQin — APPROVED** [review]
verified locally, worked as expected. just some variable names are not good.

File: include/http/middleware/prerender.js [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+            await renderResponse(req, content);
+
+        } catch(ex) {
+            // Just go normally path if there is any errors happen.
+            console.log(ex);
+        }
+    }
+});
+
+const extensionsToIgnore = [
```
Comment (CB-JackLi) [inline include/http/middleware/prerender.js:34]: minor: move it out to separate file then require it here, I believe now the file is too long

File: include/http/middleware/prerender.js [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+    '.swf',
+    '.dat',
+    '.dmg',
+    '.iso',
+    '.flv',
+    '.m4v',
+    '.torrent'
+];
+
+const crawlerUserAgents = [
```
Comment (CB-JackLi) [inline include/http/middleware/prerender.js:75]: minor: move it out to separate file then require it here, I believe now the file is too long

File: include/http/middleware/prerender.js [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+function getPrerenderedPage(req) {
+    var options = {
+        uri: url.parse(buildApiUrl(req)),
+        followRedirect: false,
+        headers: {}
+    };
+
+    Object.keys(req.headers).forEach(function (h) {
+        // Forwarding the host header can cause issues with server platforms that require it to match the URL
+        if (h == 'host') {
```
Comment (CB-JackLi) [inline include/http/middleware/prerender.js:155]: I see '!==' coming quite often, so I believe here you mean to use '===' instead of '=='

**Review by CB-JackLi — APPROVED** [review]

File: include/http/middleware/prerender.js (hunk 5 above)
Comment (vokidya) [reply include/http/middleware/prerender.js:155]: This is the code copied from third party, I will not modify too much.

**Pull request merged**